### PR TITLE
HPCC-17167 Replace dojo request

### PR DESCRIPTION
**Pull request by GordonSmith**
Fixes HPCC-17167

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -49,3 +49,6 @@
 [submodule "plugins/redis/hiredis"]
 	path = plugins/redis/hiredis
 	url = https://github.com/hpcc-systems/hiredis.git
+[submodule "esp/src/hpcc-platform-comms"]
+	path = esp/src/hpcc-platform-comms
+	url = https://github.com/GordonSmith/hpcc-platform-comms.git
```
Comment (GordonSmith) [inline .gitmodules]: TODO Switch to hpcc-systems once we have reviewed the hpcc-platform-comms API

**Comment by GordonSmith** [comment]
@miguelvazq Please review

**Comment by miguelvazq** [comment]
Looks good.

**Comment by GordonSmith** [comment]
@richardkchapman bump

File: .gitmodules (hunk 1 above)
Comment (GordonSmith) [inline .gitmodules]: Not yet - my gut tells me we should wait until the viz refactoring is complete and that the marshalle and visualizer bundle have switched over to use it as well (vs-code has switched over to use it already).

**Comment by richardkchapman** [comment]
@GordonSmith I don't want to merge until the submodule is correct. Where are we on that?

**Comment by GordonSmith** [comment]
We are no closer - once folks are freed up after the 6.4 push I plan on doing some group code reviews.
(and this PR is deliberately only the tip of the iceberg).

File: .gitmodules [hunk 2]
```diff
@@ -52,3 +52,6 @@
 [submodule "system/tbb_sm/tbb"]
     path = system/tbb_sm/tbb
     url = https://github.com/hpcc-systems/tbb.git
+[submodule "esp/src/hpcc-js-comms"]
+	path = esp/src/hpcc-js-comms
+	url = https://github.com/GordonSmith/hpcc-js-comms.git
```
Comment (GordonSmith) [inline .gitmodules]: Change to use https://github.com/hpcc-systemshpcc-js-comms.git once https://github.com/hpcc-systems/hpcc-js-comms/pull/2 is pulled.

**Comment by GordonSmith** [comment]
@miguelvazq Please review (now synced with hpcc-systems/hpcc-js-comms)

**Comment by miguelvazq** [comment]
This looks good. May need to rebase for clean merge?

**Comment by GordonSmith** [comment]
@miguelvazq Please re-review (once https://github.com/hpcc-systems/HPCC-Platform/pull/10492 is pulled and this is rebased)

**Comment by GordonSmith** [comment]
@miguelvazq - I am retesting this at the moment.

**Comment by GordonSmith** [comment]
@miguelvazq rebased and ready for review.

**Comment by miguelvazq** [comment]
Looks good.

**Comment by richardkchapman** [comment]
@GordonSmith Needs rebasing

**Comment by GordonSmith** [comment]
@richardkchapman Rebased and retested.

**Pull request merged**